standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3087,7 +3090,24 @@ private void insertTxnComponents(long txnid, LockRequest rqst, Connection dbConn
           String tblName = normalizeCase(lc.getTablename());
           String partName = normalizePartitionCase(lc.getPartitionname());
           OperationType opType = OperationType.fromDataOperationType(lc.getOperationType());
-
+          if (opType.getSqlConst().equals(OperationType.CTAS.getSqlConst())) {
```

Comment:
why do you even need to create locks for CTAS with this approach? that won't work as select doesn't even use any locking and we shouldn't add one. Why can't we stick to checkLock method and only check for Excl_Write and Exclusive locks in case of CTAS operation?

Reply:
Reverted to checkLocks method in the new commit.